app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-ecommerce-tracker.js

```diff
@@ -46,101 +43,29 @@
       }
     },
 
-    trackSearchResults: function (searchResultsBlock) {
-      var schema = this.populateEcommerceSchema(searchResultsBlock, false, null)
+    trackResults: function (searchResultsBlock) {
+      var ecommerceSchema = GOVUK.analyticsGa4.core.ecommerceHelperFunctions.populateEcommerceSchema({
+        element: searchResultsBlock,
+        resultsId: 'js-result-count'
+      })
 
-      this.clearPreviousEcommerceObject()
-      GOVUK.analyticsGa4.core.sendData(schema)
+      GOVUK.analyticsGa4.core.ecommerceHelperFunctions.clearEcommerceObject()
+      GOVUK.analyticsGa4.core.sendData(ecommerceSchema)
     },
 
     handleClick: function (event) {
-      var searchResultsBlock = event.target.closest('[data-ga4-ecommerce]')
-      var isSearchResult = event.target.getAttribute('data-ecommerce-path')
-
-      if (isSearchResult) {
-        var searchResult = event.target
-        var schema = this.populateEcommerceSchema(searchResultsBlock, true, searchResult)
+      if(event.target.getAttribute("data-ecommerce-path") !== null) {
```

Comment:
Nitpick - I think this could be simplified to `if (event.target.getAttribute("data-ecommerce-path")) {`